### ISSUE TITLE
Build with Travis, Upload to Netlify

### ISSUE THREAD

**Issue by lautarodragan** [poetapp/explorer-web#421]
We can upload the built package to Netlify instead of having Netlify build it for us.

See [Deplying to Netlify](https://www.netlify.com/docs/api/#deploying-to-netlify) and [netlify-cli](https://www.netlify.com/blog/2018/09/10/netlify-cli-2.0-now-in-beta-/).

This should allow us to use the latest version of Po.et JS in Explorer Web with minimal changes to our infrastructure.

## Deploy Previews

We may be able to mimick the deploy preview feature using _draft deploys_.

If each _draft deploy_ gets a different URL we can definitely use that for in-house deploy previews.

We'll need to add logic to the Travis build script for it to either add a comment to the PR, run as a check or whatever other way we can think of for it to share the link to the draft deploy.

## API vs. CLI

I'd default to using the CLI for simplicity and switch to the API if the CLI proves not to be meet our needs.

## Tasks

These can be broken down to separate issues.

- Research: do draft deploys get different URLs and can they be used in parallel?
- Implementation: deploy PRs to Netlify from Travis 
- Optimization: have Travis share the deploy URL on the PR 

## API Documentation

> Regarding API usage, the best ways to get familiar with our API are NOT currently in our documentation.
> 
> - from our Terms of Service, be aware of the rate limits - those are per authentication token. In case you trigger 4 deploys in a minute that 4th deploy would fail due to the 3 deploys/min rate limit.  The other API rate limits are pretty high so it's unlikely you'd hit them in normal operation.
> - unless you get officially branded as a reseller, you'd be unable to manage users - that (and managing payments/ subscriptions) are basically the only things you cannot do as a non-reseller using our API.
> - https://open-api.netlify.com has a fairly thorough list of the available calls but is not entirely fleshed out or complete as it has to be manually updated with new calls.  So...
> - checking how our UI actually uses the API in your browser dev tools will be the best guide to using every feature you can control with the API.  I realize this is a bit inelegant but it is definitely the best way to figure out what structure a Repo object has or what the normal "order of operations" is in creating a site.

**Comment by lautarodragan** [comment]
Confirmation from support: 


> Yes, you can do the build and upload using either our [netlify-cli](https://www.netlify.com/blog/2018/09/10/netlify-cli-2.0-now-in-beta-/) or our API.
> 
> Regarding API usage, the best ways to get familiar with our API are NOT currently in our documentation.
> 
> - from our Terms of Service, be aware of the rate limits - those are per authentication token. In case you trigger 4 deploys in a minute that 4th deploy would fail due to the 3 deploys/min rate limit.  The other API rate limits are pretty high so it's unlikely you'd hit them in normal operation.
> - unless you get officially branded as a reseller, you'd be unable to manage users - that (and managing payments/ subscriptions) are basically the only things you cannot do as a non-reseller using our API.
> - https://open-api.netlify.com has a fairly thorough list of the available calls but is not entirely fleshed out or complete as it has to be manually updated with new calls.  So...
> - checking how our UI actually uses the API in your browser dev tools will be the best guide to using every feature you can control with the API.  I realize this is a bit inelegant but it is definitely the best way to figure out what structure a Repo object has or what the normal "order of operations" is in creating a site.

Waiting for reply on how to deploy as a PR preview.

**Comment by lautarodragan** [comment]
Regarding deploy previews:

> You cannot do exactly what you ask, Lautaro. Pull requests are only from git-backed sites.
>
> You can still do a "draft" deploy that isn't published at your production URL, via either API or CLI.  CLI is easier, you run `netlify deploy` and by default it is a draft deploy. 

If each _draft deploy_ gets a different URL we can definitely use that for in-house deploy previews.

We'll need to add logic to the Travis build script for it to either add a comment to the PR, run as a check or whatever other way we can think of for it to share the link to the draft deploy.

**Comment by lautarodragan** [comment]
Note: we don't need this implemented in explorer-web or frost-web right now. We'll probably address this in the new hybrid site directly.

**Comment by geoffturk** [comment]
OK. I've parked this in the PNE epic as a reminder.

**Comment by geoffturk** [comment]
I spoke with Jesse about this and he is happy to try dev deployments on now.sh (Zeit).
https://zeit.co/now

**Comment by geoffturk** [comment]
Closing this as @jwicks31 will ensure poet-js is setup correctly on new platform.
